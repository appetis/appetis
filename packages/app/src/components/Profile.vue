<template>
  <div
    class="fixed z-10 inset-0 overflow-y-auto"
    aria-labelledby="modal-title"
    role="dialog"
    aria-modal="true"
  >
    <div class="modal">
      <div class="modal-background" aria-hidden="true"></div>

      <!--      <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">&#8203;</span>-->

      <div class="modal-content-md">
        <span class="modal-close" @click="close">
          <vue-feather size="15" type="x"></vue-feather>
        </span>
        <div class="modal-body">
          <div class="sm:flex sm:items-start">
            <div class="profile-photo">
              <img :src="displayImage" />
              <div class="profile-online"></div>
            </div>
            <div class="profile-top">
              <h3
                class="text-md leading-6 font-medium text-gray-900"
                id="profile-name"
              >
                Susan Guerrero
              </h3>
              <div class="mt-0">
                <p class="text-xs text-gray-500" id="profile-position">
                  Project Manager
                </p>
              </div>
              <div class="mt-3">
                <div class="profile-icon">
                  <vue-feather
                    size="14"
                    class="mt-1.5"
                    type="calendar"
                  ></vue-feather>
                </div>
                <div class="profile-icon">
                  <vue-feather
                    size="14"
                    class="mt-1.5"
                    type="mail"
                  ></vue-feather>
                </div>
                <div class="profile-icon">
                  <vue-feather
                    size="14"
                    class="mt-1.5"
                    type="message-square"
                  ></vue-feather>
                </div>
              </div>
            </div>
          </div>
          <div class="mt-8 text-sm">
            <div class="mb-3 inline-block w-full">
              <vue-feather
                size="17"
                class="text-gray-400 align-middle"
                type="map-pin"
              ></vue-feather>
              <span class="ml-3 align-middle"> In office </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <vue-feather
                size="17"
                class="text-gray-400 align-middle"
                type="user"
              ></vue-feather>
              <span class="ml-3 align-middle">
                Team/department - project manager
              </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <vue-feather
                size="17"
                class="text-gray-400 align-middle"
                type="mail"
              ></vue-feather>
              <span class="ml-3 align-middle"> random@email.com </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <vue-feather
                size="17"
                class="text-gray-400 align-middle"
                type="phone"
              ></vue-feather>
              <span class="ml-3 align-middle"> Ext.335 </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <vue-feather
                size="17"
                class="text-gray-400 align-middle"
                type="smartphone"
              ></vue-feather>
              <span class="ml-3 align-middle"> 647-999-1234 </span>
            </div>
          </div>
        </div>
        <!--        <div class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">-->
        <!--          <button type="button" class="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm">-->
        <!--            Deactivate-->
        <!--          </button>-->
        <!--          <button type="button" @click="close" class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm">-->
        <!--            Cancel-->
        <!--          </button>-->
        <!--        </div>-->
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Profile',
  props: {
    avatar: {
      required: true,
      type: String,
    },
  },
  data() {
    return {}
  },
  methods: {
    close() {
      this.$emit('close')
    },
  },
  computed: {
    displayImage() {
      return this.avatar
    },
  },
}
</script>

<style scoped></style>
