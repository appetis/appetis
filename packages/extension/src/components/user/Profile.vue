<template>
  <div
    class="fixed inset-0 overflow-y-auto z-10"
    aria-labelledby="modal-title"
    role="dialog"
    aria-modal="true"
  >
    <div class="modal-background" aria-hidden="true"></div>
    <div class="modal">
      <!--      <span class="hidden sm:inline-block sm:align-middle sm:h-screen" aria-hidden="true">&#8203;</span>-->

      <div class="modal-content-sm">
        <span class="modal-close" @click="close">
          <v-icon name="x" base-class="v-icon"></v-icon>
        </span>
        <div class="modal-body" v-if="profile">
          <div>
            <Avatar :Profile="profile.Profile" />
            <div class="profile-top">
              <h3
                class="text-lg leading-6 font-medium text-gray-900"
                id="profile-name"
              >
                {{ profile.Profile.name }}
              </h3>
              <div class="mt-0">
                <span
                  class="
                    inline-block
                    bg-wk-defgray
                    rounded-sm
                    px-2
                    mr-2
                    text-white
                  "
                >
                  Admin
                </span>
                <p
                  class="
                    inline-block
                    text-sm text-brand-defgray
                    profile-position
                  "
                >
                  {{ profile.Profile.position }}
                </p>
              </div>

              <!-- // For version 2
              <div class="mt-3">
                <div class="profile-icon">
                  <v-icon name="calendar" base-class="vb-icon"></v-icon>
                </div>
                <div class="profile-icon">
                  <v-icon name="mail" base-class="vb-icon"></v-icon>
                </div>
                <div class="profile-icon">
                  <v-icon name="message-square" base-class="vb-icon"></v-icon>
                </div>
              </div>
              -->
            </div>
          </div>
          <div class="mt-8 text-sm">
            <div class="mb-3 inline-block w-full">
              <span class="w-18">
                <v-icon
                  name="map-pin"
                  base-class="pv-icon"
                  class="text-brand-defgray"
                ></v-icon>
              </span>
              <span class="ml-3 align-middle" id="profile-status">
                In office
              </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <v-icon
                name="user"
                base-class="pv-icon"
                class="text-brand-defgray"
              ></v-icon>
              <span class="ml-3 align-middle">
                <span id="profile-department">{{
                  profile.Profile.department
                }}</span>
                -
                <span class="profile-position">{{
                  profile.Profile.position
                }}</span>
              </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <v-icon
                name="mail"
                base-class="pv-icon"
                class="text-brand-defgray"
              ></v-icon>
              <span class="ml-3 align-middle" id="profile-email">
                {{ profile.email }}
              </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <v-icon
                name="phone"
                base-class="pv-icon"
                class="text-brand-defgray"
              ></v-icon>
              <span class="ml-3 align-middle">
                Ext.
                <span id="profile-phone-ext">{{
                  profile.Profile.phone_ext
                }}</span>
              </span>
            </div>
            <div class="mb-3 inline-block w-full">
              <v-icon
                name="smartphone"
                base-class="pv-icon"
                class="text-brand-defgray"
              ></v-icon>
              <span class="ml-3 align-middle" id="profile-phone">
                {{ profile.Profile.phone }}
              </span>
            </div>
          </div>
        </div>
        <!--        <div class="bg-gray-50 px-4 py-3 sm:px-6 sm:flex sm:flex-row-reverse">-->
        <!--          <button type="button" class="w-full inline-flex justify-center rounded-md border border-transparent shadow-sm px-4 py-2 bg-red-600 text-base font-medium text-white hover:bg-red-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-red-500 sm:ml-3 sm:w-auto sm:text-sm">-->
        <!--            Deactivate-->
        <!--          </button>-->
        <!--          <button type="button" @click="close" class="mt-3 w-full inline-flex justify-center rounded-md border border-gray-300 shadow-sm px-4 py-2 bg-white text-base font-medium text-gray-700 hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm">-->
        <!--            Cancel-->
        <!--          </button>-->
        <!--        </div>-->
      </div>
    </div>
  </div>
</template>

<script>
import Avatar from './Avatar'
export default {
  name: 'Profile',
  props: {
    profile: null,
  },
  components: {
    Avatar,
  },
  data() {
    return {}
  },
  methods: {
    close() {
      this.$emit('close')
    },
  },
}
</script>
<style scoped>
.pv-icon {
  width: 20px;
}
.vb-icon {
  width: 22px;
}
</style>
