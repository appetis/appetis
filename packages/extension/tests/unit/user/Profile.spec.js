import Profile from '@/components/user/Profile'
import { mount } from '@vue/test-utils'
import { mockUser } from '../../data'

const factory = () => {
  return mount(Profile, {
    propsData: {
      profile: mockUser().user,
    },
  })
}

describe('init/Profile', () => {
  it('profile modal snapshot', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('display profile data', () => {
    const wrapper = factory()
    const user = mockUser().user

    expect(wrapper.find('#profile-name').text()).toBe(user.Profile.name)
    expect(wrapper.find('.profile-position').text()).toBe(user.Profile.position)
    expect(wrapper.find('#profile-department').text()).toBe(
      user.Profile.department
    )
    expect(wrapper.find('#profile-email').text()).toBe(user.email)
    expect(wrapper.find('#profile-phone-ext').text()).toBe(
      user.Profile.phone_ext
    )
    expect(wrapper.find('#profile-phone').text()).toBe(user.Profile.phone)
  })

  it('click close button', () => {
    const wrapper = factory()
    const mockClose = jest.fn()
    wrapper.vm.$on('close', mockClose)
    wrapper.find('.modal-close').trigger('click')

    expect(mockClose).toHaveBeenCalledTimes(1)
  })
})
