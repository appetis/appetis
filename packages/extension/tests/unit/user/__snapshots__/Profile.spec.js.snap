// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init/Profile profile modal snapshot 1`] = `
<div aria-labelledby="modal-title" role="dialog" aria-modal="true" class="fixed inset-0 overflow-y-auto z-10">
  <div aria-hidden="true" class="modal-background"></div>
  <div class="modal">
    <div class="modal-content-sm"><span class="modal-close"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="v-icon icon-x"><line x1="18" x2="6" y1="6" y2="18"></line><line x1="6" x2="18" y1="6" y2="18"></line></svg></span>
      <div class="modal-body">
        <div>
          <div class="profile-photo">
            <div class="header-no-avatar"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="profile-no-icon icon-user">
                <path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"></path>
                <circle cx="12" cy="7" r="4"></circle>
              </svg></div> <img src="" class="profile-avatar" style="display: none;">
          </div>
          <div class="profile-top">
            <h3 id="profile-name" class="text-lg leading-6 font-medium text-gray-900">
              Hyewon Kim
            </h3>
            <div class="mt-0"><span class="
                  inline-block
                  bg-wk-defgray
                  rounded-sm
                  px-2
                  mr-2
                  text-white
                ">
                Admin
              </span>
              <p class="inline-block text-sm text-brand-defgray profile-position">
                Developer
              </p>
            </div>
          </div>
        </div>
        <div class="mt-8 text-sm">
          <div class="mb-3 inline-block w-full"><span class="w-18"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="text-brand-defgray pv-icon icon-map-pin"><path d="M21 10c0 7-9 13-9 13s-9-6-9-13a9 9 0 0 1 18 0z"></path><circle cx="12" cy="10" r="3"></circle></svg></span> <span id="profile-status" class="ml-3 align-middle">
              In office
            </span></div>
          <div class="mb-3 inline-block w-full"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="text-brand-defgray pv-icon icon-user">
              <path d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"></path>
              <circle cx="12" cy="7" r="4"></circle>
            </svg> <span class="ml-3 align-middle"><span id="profile-department">IT</span> -
            <span class="profile-position">Developer</span></span>
          </div>
          <div class="mb-3 inline-block w-full"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="text-brand-defgray pv-icon icon-mail">
              <path d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"></path>
              <polyline points="22,6 12,13 2,6"></polyline>
            </svg> <span id="profile-email" class="ml-3 align-middle">
              hyewon@workly.page
            </span></div>
          <div class="mb-3 inline-block w-full"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="text-brand-defgray pv-icon icon-phone">
              <path d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"></path>
            </svg> <span class="ml-3 align-middle">
              Ext.
              <span id="profile-phone-ext">222</span></span></div>
          <div class="mb-3 inline-block w-full"><svg viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="text-brand-defgray pv-icon icon-smartphone">
              <rect x="5" y="2" width="14" height="20" rx="2" ry="2"></rect>
              <line x1="12" x2="12.01" y1="18" y2="18"></line>
            </svg> <span id="profile-phone" class="ml-3 align-middle">
              222-222-2222
            </span></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
